filecheck/parser.py

```diff
@@ -43,6 +53,10 @@ def pattern_for_opts(opts: Options) -> tuple[re.Pattern[str], re.Pattern[str]]:
     r"\[\[#(\$?[a-zA-Z_][a-zA-Z0-9_]*)([a-z0-9 +\-()]*)]]"
 )
 
+# pseudo numeric variable, see:
+# https://llvm.org/docs/CommandGuide/FileCheck.html#filecheck-pseudo-numeric-variables
+PSEUDO_NUMERIC_VARIABLE = re.compile(r"\[\[# @line( ([+-]{1}) (\d+))?]]")
```

Comment:
The LLVM format is `@LINE` not `@line`, so we need to uppercase that here and in the tests.

I also have two nits:

nit: `{1}` can be omitted here, I think? 
supernit: I think this does not accept `[[# @line ]`` (with a trailing space), which FileCheck accepts